src/Tinker.jl

```diff
@@ -1,4 +1,268 @@
 module Tinker
 
-# package code goes here
+using Gtk.ShortNames, GtkReactive, Graphics, Colors, Images
+
+## Generally useful structs and functions
+# Rectangle structure
+struct Rectangle
+    x::Float64
+    y::Float64
+    w::Float64
+    h::Float64
+end
+
+Rectangle() = Rectangle(0,0,-1,-1)
+Base.isempty(R::Rectangle) = R.w < 0 || R.h < 0
+
+# rectangle draw function
+function drawrect(ctx, rect, color)
+    set_source(ctx, color)
+    rectangle(ctx, rect.x, rect.y, rect.w, rect.h)
+    stroke(ctx)
+end;
+
+# Zooms zr to the decimal % entered; view centered around center XY
+function zoom_percent(z::Float64, zr::ZoomRegion, center::XY{Int64})
```

Comment:
Also the `Int64` signature is too specific because it would fail on an older 32-bit CPU. `Int` works for both 64-bit and 32-bit machines.